gprofiler/profilers/java.py

```diff
@@ -259,11 +285,109 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
+        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid, "jvm_flags": self.get_jvm_flags(process)}
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    @functools.lru_cache(maxsize=1024)
+    def get_jvm_flags(self, process: Process) -> Optional[List[tuple]]:
+        jvm_raw_flags = self.get_jvm_flags_raw(process)
+        if jvm_raw_flags is None:
+            return None
+
+        return [dataclasses.astuple(flag) for flag in jvm_raw_flags if self.filter_jvm_flags(flag)]
+
+    @staticmethod
+    def filter_jvm_flags(flag: JvmFlag) -> bool:
+        if flag.origin == "default":
+            return False
+
+        if flag.type not in [
+            "bool",
+            "int",
+            "uint",
+            "intx",
+            "uintx",
+            "uint64_t",
+            "size_t",
+            "double",
+        ]:
+            return False
+
+        if set(flag.kind).intersection({"notproduct", "pd", "develop"}):
```

Comment:
```suggestion
        if flag.kind not in ("notproduct", "pd", "develop"):
```
? isn't it a single string?

Reply:
Or kind is a list of strings, let's name it `kinds` not `kind`?

Reply:
Kind can be both `product` and `C2`, for example.

Reply:
opposite of @YoniKF's comment, I consider storing flag.kind as `set`

Reply:
I see, it's a collection. Closing